product/roundhouse.tasks/Roundhouse.cs

```diff
@@ -142,12 +153,14 @@ public string RecoveryMode
             set
             {
                 RecoveryMode result;
-                if (!Enum.TryParse(value, true, out result))
+                if (Enum.TryParse(value, true, out result))
                 {
-                    throw new ArgumentOutOfRangeException("value", value, "The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'.");
+                    this.recoveryMode = result;
+                }
+                else
+                {
+                    this.loggingHelper.LogError("The value of 'RecoveryMode' must be one of these values: 'NoChange', 'Simple' or 'Full'. Actual value was '{0}'.", value);

```

Comment:
How do I get a reference to the logger? I looked at the IoC container stuff that is already built in, but couldn't make much sense of it.